### PR TITLE
fix nsector calculation

### PR DESCRIPTION
**Pull request by petemoore**
This is a minor optimisation. The calculation for the number of sectors required to store a file overestimated the number of sectors required by one when the filelength+9 was exactly a multiple of 510.

The previous calculation was:
```py
nsectors = 1 + (filelength+9)//510
```
when considering, e.g. a file length of 501 bytes, the total storage needed is 510 bytes, which requires only one sector. However, with this calculation, `nsectors` would be `1 + (501+9)//510 = 1+ 510//510 = 2`.

I've changed the calculation to
```py
nsectors = (filelength+9+509)//510
```
which gives in the same example, `(501+9+509)//510 = (510+509)//510 = 1`, and as soon as one byte of storage is added, it would roll over to 2 sectors, e.g. `(502+9+509)//510 = (511+509)//510 = (510+510)//510 = 2`.

In other words, in the rare case that the filelength+9 is a multiple of 510 bytes, it will save one sector.

### CODE REVIEW

**Comment by stefandrissen** [comment]
Instead of three lines explaining the calculation  edge case, I would eliminate 0 with max:

nsectors = max( 1, ( filelength + 9 ) // 510 )

**Comment by petemoore** [comment]
Thanks Stefan. You make a good point - the comment is rather long. Should I trim/remove it?

> I would eliminate 0 with max:
> 
> nsectors = max( 1, ( filelength + 9 ) // 510 )

Unfortunately I think this won't work when filelength > 501. Consider e.g. filelength = 600, which requires 2 sectors, but the formula returns 1.

**Comment by stefandrissen** [comment]
Good spot Pete, reminder to self: never attempt to code on a phone 🤦

**Comment by petemoore** [comment]
Heh, no worries! A more concise version could be:

```python
    # round sectors "up" not "down"
    nsectors = 1 + (filelength+9-1)//510
```

I'm not sure if the comment sufficiently explains the formula, but I'm happy to go with either version.

**Comment by simonowen** [comment]
Any takers for this to make the rounding up more explicit?

```python
nsectors = math.ceil(( filelength + 9 ) / 510)
```

I'd probably omit the comment too as the variable name is at least as descriptive as the surrounding code.

**Comment by petemoore** [comment]
I like it! Good old python, didn't know about math.ceil. :-) Thanks Simon!

**Comment by simonowen** [comment]
Thanks Pete!
